stores/script/src/main/java/com/bloxbean/cardano/yaci/store/script/helper/TxScriptFinder.java

```diff
@@ -63,13 +65,15 @@ public Map<String, PlutusScript> getScripts(Transaction transaction) {
                     .forEach(addressUtxoOptional -> {
                         AddressUtxo addressUtxo = addressUtxoOptional.get();
                         if (addressUtxo.getScriptRef() != null && !addressUtxo.getScriptRef().isEmpty()) {
-                            PlutusScript plutusScript = ScriptUtil.deserializeScriptRef(addressUtxo);
-                            if (plutusScript != null)
-                                scriptsMap.put(ScriptUtil.getPlutusScriptHash(plutusScript), plutusScript);
-                            else {
-                                log.error("PlutusScript is null for tx: " + transaction.getTxHash());
-                            }
-                        }
+                            var cclScript = ScriptReferenceUtil.deserializeScriptRef(HexUtil.decodeHexString(addressUtxo.getScriptRef()));
```

Comment:
@Sotatek-HuyLe3a What would happen if `deserializeScriptRef` throws an error? Ideally, we should just log the error and continue. Can you please verify?

Can we avoid double deserialization here? Since `ScriptReferenceUtil` is already deserializing to CCL's `PlutusScript`, perhaps we only need to handle the conversion to the store's `PlutusScript` inside a new conversion method in `ScriptUtil`